### PR TITLE
✨ Source Monday: Add incremental sync for Items stream

### LINKED ISSUE #20950: Source Monday.com: Incremental Sync
## Tell us about the problem you're trying to solve
The Monday source ([repo](https://github.com/airbytehq/airbyte/tree/master/airbyte-integrations/connectors/source-monday), [docs](https://docs.airbyte.com/integrations/sources/monday/)) currently implements Full Refresh Sync mode only: Airbyte users must trigger the retrieval of  _all records_ in a Monday account upon every sync.

The **addition of an Incremental Sync mode** would provide Airbyte users with a more efficient and more satisfactory solution for Monday accounts with a large number of items and/or a requirement for a frequent sync interval, for which full refresh may be unacceptable.

This may also help resolve #15569 and [related issues](https://github.com/airbytehq/airbyte/labels/connectors%2Fsource%2Fmonday).

Implementation of an incremental stream for the [Monday GraphQL API](https://developer.monday.com/api-reference/docs/items-queries) initially appears to be not possible, or at least not to be as straightforward as it might be with some other APIs, given the absence of an `updated_after` or other datetime argument for most queries: users of the API are seemingly [expected to request all data](https://community.monday.com/t/updated-since-filter/42545/4) without use of a time-based cursor. However, there is an available solution, described below.

## Describe the solution you’ve considered
Of the objects/streams available, board `items` (previously named `pulse` in Monday) is perhaps the one most likely to grow in scale to a large number of records, and so incremental sync of items in particular would be most useful (whereas the boards, users and teams are less likely to greatly increase as usage increases). 

### The Activity Logs query can return items updated after given time
An indirect strategy for retrieving items incrementally is available via the [Activity Logs query](https://developer.monday.com/api-reference/docs/activity-logs#understanding-the-format-of-the-created_at-field):

- The `from` argument accepts a timestamp that could be used for incremental sync
- For each item in the array/list returned where `"entity": "pulse"`:
  - The `data` field returns a JSON string which includes `{"pulse": {"id": 4916413449}}`
  - Given that pulse is a legacy term for item, the pulse id is the item id
- Therefore by parsing the response JSON and unpacking the item ids, it is possible to reliably obtain a list of the ids of all items updated since the timestamp argument given

So, upon each incremental sync, an Airbyte source could first query activity logs using the timestamp from the last sync, then use the list of item IDs in the `ids` argument of the [Items query](https://developer.monday.com/api-reference/docs/items-queries) with a [substream slicer](https://docs.airbyte.com/connector-development/config-based/understanding-the-yaml-file/stream-slicers/#substreamslicer). Incremental sync of items would then be possible, while full sync for the other streams will likely remain acceptable as currently implemented.

## Are you willing to submit a PR?

Without significant support and a deep understanding of Airbyte development, I would not feel confident in correctly implementing the logic described above.

I've attempted to modify the existing low-code and Python CDK versions, but have not been able to find enough examples and docs to make clear which approach I should use and where Airbyte's abstractions accommodate my goal versus where they should be overridden.

I was hoping to implement this myself within a week, but I currently feel more confident that I would be able to achieve that by building an ETL script with Python from scratch, than I do about being able to understanding enough about [which parts of the CDK I should work with](https://airbyte-cdk.readthedocs.io/en/stable/api/airbyte_cdk.sources.html). I feel that after investing significant time into getting to know the CDK, I could still find myself a long way away from being able to implement what I've described above. So, opening this request and making a start on my own solution in the meantime seems the favourable approach, but I'd be open to reconsidering!

While Airbyte is actively evolving and adapting, the recent and ongoing state of transition from CDK to config/YAML (#19586, #20533, which I recognise is necessary for continual improvement) makes it difficult for me as an outsider to know which framework would be suitable at this point in time. Perhaps a developer who is involved and experienced in this work would be better placed to investigate the strategy proposed above. I would be happy to help with testing or less technical contributions if useful. Many thanks for your help!

### PR DESCRIPTION
**Pull request by tolik0**
## What
Implements incremental sync for Items stream. Adds new stream: Activity logs. 
#20950 

## How

This pull request implements incremental synchronization for the Items stream by implementing the approach described in the connected issue.  The implementation follows these steps:

- Initial Sync: All items are synced initially without relying on activity logs.
- Subsequent Sync: The process involves fetching updates since a specified date from the Activity logs stream. The modified item IDs are extracted from the activity logs. Based on these extracted IDs, the corresponding items are selectively synced to ensure their data is up to date.

Following this process, the pull request achieves incremental synchronization by selectively updating only the modified items, as indicated by the activity logs.

## Recommended reading order
1. `source-monday/manifest.yaml`
2. `source-monday/extractor.py`
3. `source-monday/components.py`
4. `source-monday/graphql_requester.py`

### CODE REVIEW

File: airbyte-integrations/connectors/source-monday/integration_tests/sample_state.json [hunk 1]
```diff
@@ -1,5 +1,27 @@
...
+      "stream_descriptor" : {
+        "name" : "activity_logs"
+      },
+      "stream_state" : {
+        "created_at_int" : 1619041749
+      }
+    }
   }
-}
+]
```
Comment (bazarnov) [inline airbyte-integrations/connectors/source-monday/integration_tests/sample_state.json:27]: nit: add new line

File: airbyte-integrations/connectors/source-monday/source_monday/components.py [hunk 2]
```diff
@@ -0,0 +1,401 @@
...
+from airbyte_cdk.sources.declarative.partition_routers.substream_partition_router import ParentStreamConfig
+from airbyte_cdk.sources.declarative.requesters.request_option import RequestOptionType
+from airbyte_cdk.sources.declarative.stream_slicers.stream_slicer import StreamSlicer
+from airbyte_cdk.sources.declarative.types import Config, Record, StreamSlice, StreamState
+from airbyte_cdk.sources.streams.core import Stream
+
+RequestInput = Union[str, Mapping[str, str]]
+
+
+def is_valid_format(date_string):
```
Comment (bazarnov) [inline airbyte-integrations/connectors/source-monday/source_monday/components.py:21]: typings

File: airbyte-integrations/connectors/source-monday/source_monday/components.py [hunk 3]
```diff
@@ -0,0 +1,401 @@
...
+    try:
+        datetime.strptime(date_string, "%Y-%m-%dT%H:%M:%SZ")
+        return True
+    except ValueError:
+        return False
+
+
+def get_max_date_string(str1, str2):
```
Comment (bazarnov) [inline airbyte-integrations/connectors/source-monday/source_monday/components.py:29]: typings

File: airbyte-integrations/connectors/source-monday/source_monday/components.py [hunk 4]
```diff
@@ -0,0 +1,401 @@
...
+                        if substream_slice not in all_ids:
+                            all_ids.add(substream_slice)
+                            slice[self.substream_slice_field].append(substream_slice)
+                        slice[self.cursor_field.eval(self.config)] = self._cursor.get(self.cursor_field.eval(self.config))
+                        slice[self.parent_stream_name] = {
+                            self.parent_cursor_field: self._cursor.get(self.parent_stream_name, {}).get(self.parent_cursor_field)
+                        }
+                        self._update_parent_cursor(slice, parent_record)
+
+                        print(f"{slice=}")
```
Comment (bazarnov) [inline airbyte-integrations/connectors/source-monday/source_monday/components.py:381]: this should not be present

File: airbyte-integrations/connectors/source-monday/source_monday/graphql_requester.py [hunk 5]
```diff
@@ -104,6 +137,16 @@ def get_request_params(
             query_builder = partial(self._build_items_query, sub_page=sub_page)
         elif self.name == "teams":
             query_builder = self._build_teams_query
+        elif self.name == "activity_logs":
+            page, sub_page = page if page else (None, None)
+            print(f"{stream_slice=} {stream_state=}")
```
Comment (bazarnov) [inline airbyte-integrations/connectors/source-monday/source_monday/graphql_requester.py:142]: remove print

File: airbyte-integrations/connectors/source-monday/source_monday/manifest.yaml [hunk 6]
```diff
@@ -54,6 +54,12 @@ definitions:
       type: "PageIncrement"
       start_from_page: 1
       page_size: 100
+  double_paginator:
```
Comment (bazarnov) [inline airbyte-integrations/connectors/source-monday/source_monday/manifest.yaml:57]: does it mean "nested" pagination?

File: airbyte-integrations/connectors/source-monday/source_monday/manifest.yaml [hunk 7]
```diff
@@ -111,14 +122,92 @@ definitions:
...
   - "#/definitions/updates_stream"
   - "#/definitions/users_stream"
+  - "#/definitions/workspaces_stream"
+  - "#/definitions/items_incremental_stream"
+  - "#/definitions/activity_logs_stream"
 
 check:
   stream_names:
-    - "users"
+    - "users"
```
Comment (bazarnov) [inline airbyte-integrations/connectors/source-monday/source_monday/manifest.yaml:213]: nit: new line should be added

File: airbyte-integrations/connectors/source-monday/source_monday/schemas/tags.json [hunk 8]
```diff
@@ -0,0 +1,9 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "type": "object",
+  "properties": {
+    "color": { "type": ["null", "string"] },
+    "id": { "type": ["null", "integer"] },
+    "name": { "type": ["null", "string"] }
+  }
+}
```
Comment (bazarnov) [inline airbyte-integrations/connectors/source-monday/source_monday/schemas/tags.json:9]: new line

File: airbyte-integrations/connectors/source-monday/source_monday/schemas/workspaces.json [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+      "items": {
+        "type": ["null", "object"],
+        "additionalProperties": true,
+        "properties": {
+          "id": { "type": ["null", "integer"] }
+        }
+      }
+    }
+  }
+}
```
Comment (bazarnov) [inline airbyte-integrations/connectors/source-monday/source_monday/schemas/workspaces.json:73]: new line

**Review by bazarnov — CHANGES_REQUESTED** [review]
small changes are requested, also please run the CAT locally before making this PR "ready for review"

thanks

File: airbyte-integrations/connectors/source-monday/source_monday/components.py (hunk 2 above)
Comment (tolik0) [reply airbyte-integrations/connectors/source-monday/source_monday/components.py]: Done

File: airbyte-integrations/connectors/source-monday/source_monday/components.py (hunk 4 above)
Comment (tolik0) [reply airbyte-integrations/connectors/source-monday/source_monday/components.py]: Done

File: airbyte-integrations/connectors/source-monday/source_monday/graphql_requester.py (hunk 5 above)
Comment (tolik0) [reply airbyte-integrations/connectors/source-monday/source_monday/graphql_requester.py]: done

File: airbyte-integrations/connectors/source-monday/source_monday/components.py (hunk 3 above)
Comment (tolik0) [reply airbyte-integrations/connectors/source-monday/source_monday/components.py]: Done

File: airbyte-integrations/connectors/source-monday/source_monday/manifest.yaml (hunk 6 above)
Comment (tolik0) [reply airbyte-integrations/connectors/source-monday/source_monday/manifest.yaml:57]: Yes

**Comment by octavia-squidington-iii** [comment]
##  source-monday test report (commit [`abde7c925e`](https://github.com/airbytehq/airbyte/pull/27944/commits/abde7c925edfa0aab7e43057ffe0b8f1a4516a0a)) - ❌

⏲️  Total pipeline duration: 01mn31s 

| Step                                                                 | Result   |
|:---------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-monday/metadata.yaml | ✅        |
| Connector version semver check                                       | ✅        |
| QA checks                                                            | ✅        |
| Code format checks                                                   | ✅        |
| Connector package install                                            | ✅        |
| Build source-monday docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                           | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/tolik0_source-monday_20950-source-mondaycom-incremental-sync/1688746312/abde7c925edfa0aab7e43057ffe0b8f1a4516a0a/source-monday/1.1.0/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-monday test
```

**Comment by octavia-squidington-iii** [comment]
##  source-monday test report (commit [`7b7c9abcf5`](https://github.com/airbytehq/airbyte/pull/27944/commits/7b7c9abcf53bf291e8b63b755376569e7559fe2a)) - ❌

⏲️  Total pipeline duration: 19mn38s 

| Step                                                                 | Result   |
|:---------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-monday/metadata.yaml | ✅        |
| Connector version semver check                                       | ✅        |
| QA checks                                                            | ✅        |
| Code format checks                                                   | ❌        |
| Connector package install                                            | ✅        |
| Build source-monday docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                           | ✅        |
| Acceptance tests                                                     | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/tolik0_source-monday_20950-source-mondaycom-incremental-sync/1688748416/7b7c9abcf53bf291e8b63b755376569e7559fe2a/source-monday/1.1.0/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-monday test
```

**Comment by octavia-squidington-iii** [comment]
##  source-monday test report (commit [`16c5918f8b`](https://github.com/airbytehq/airbyte/pull/27944/commits/16c5918f8bd9819c3fe58909e0c05fed4ff12f3c)) - ✅

⏲️  Total pipeline duration: 17mn48s 

| Step                                                                 | Result   |
|:---------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-monday/metadata.yaml | ✅        |
| Connector version semver check                                       | ✅        |
| QA checks                                                            | ✅        |
| Code format checks                                                   | ✅        |
| Connector package install                                            | ✅        |
| Build source-monday docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                           | ✅        |
| Acceptance tests                                                     | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/tolik0_source-monday_20950-source-mondaycom-incremental-sync/1688753342/16c5918f8bd9819c3fe58909e0c05fed4ff12f3c/source-monday/1.1.0/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-monday test
```

**Review by bazarnov — APPROVED** [review]

File: airbyte-integrations/connectors/source-monday/source_monday/components.py [hunk 10]
```diff
@@ -0,0 +1,252 @@
...
+    def get_stream_state(self) -> StreamState:
+        return self._state
+
+    def _get_max_state_value(
+        self,
+        current_state_value: Optional[Union[int, str]],
+        last_record_value: Optional[Union[int, str]],
+    ) -> Optional[Union[int, str]]:
+        if isinstance(last_record_value, str) and isinstance(current_state_value, str):
+            return compare_date_strings(current_state_value, last_record_value)
```
Comment (lazebnyi) [inline airbyte-integrations/connectors/source-monday/source_monday/components.py:96]: We can perform string comparisons without converting the type to datetime.

File: airbyte-integrations/connectors/source-monday/source_monday/components.py [hunk 11]
```diff
@@ -0,0 +1,252 @@
...
+            return
+        self._state[self.cursor_field.eval(self.config)] = max_dt[self.cursor_field.eval(self.config)]
+
+        if self.parent_stream:
+            if self.parent_cursor_field not in most_recent_record:
+                date_str = most_recent_record[self.cursor_field.eval(self.config)]
+                parent_cursor = int(datetime.strptime(date_str, "%Y-%m-%dT%H:%M:%S%z").timestamp())
+            else:
+                parent_cursor = most_recent_record[self.parent_cursor_field]
+            self._state[self.parent_stream_name] = {self.parent_cursor_field: parent_cursor}
```
Comment (lazebnyi) [inline airbyte-integrations/connectors/source-monday/source_monday/components.py:197]:
```suggestion
    def close_slice(self, stream_slice: StreamSlice, last_record: Optional[Record]) -> None:
        max_dt = self._state if self.is_greater_than_or_equal(self._state, most_recent_record) else most_recent_record

        if not max_dt:
            return

        cursor_field_eval = self.cursor_field.eval(self.config)
        self._state[cursor_field_eval] = max_dt[cursor_field_eval]

        if self.parent_stream:
            date_str = most_recent_record[self.cursor_field.eval(self.config)]
            default_parent_cursor = int(datetime.strptime(date_str, "%Y-%m-%dT%H:%M:%S%z").timestamp())
            
            parent_cursor = most_recent_record.get(self.parent_cursor_field, default_parent_cursor)

            self._state[self.parent_stream_name] = {self.parent_cursor_field: parent_cursor}
```

File: airbyte-integrations/connectors/source-monday/source_monday/components.py [hunk 12]
```diff
@@ -0,0 +1,252 @@
...
+                yield from []
+
+    def stream_slices(self) -> Iterable[Mapping[str, Any]]:
+        parent_state = (self._state or {}).get(self.parent_stream_name, {})
+
+        slices_generator = self.read_parent_stream(self.parent_sync_mode, self.parent_cursor_field, parent_state)
+        if self.parent_complete_fetch:
+            yield from [slice for slice in slices_generator]
+        else:
+            yield from slices_generator
```
Comment (lazebnyi) [inline airbyte-integrations/connectors/source-monday/source_monday/components.py:252]:
```suggestion
        yield from [slice for slice in slices_generator] if self.parent_complete_fetch else slices_generator
```

File: airbyte-integrations/connectors/source-monday/source_monday/extractor.py [hunk 13]
```diff
@@ -0,0 +1,88 @@
...
+            if "*" in path:
+                extracted = dpath.util.values(response_body, path)
+            else:
+                extracted = dpath.util.get(response_body, path, default=[])
+        if isinstance(extracted, list):
+            return extracted
+        elif extracted:
+            return [extracted]
+        else:
+            return []
```
Comment (lazebnyi) [inline airbyte-integrations/connectors/source-monday/source_monday/extractor.py:82]:
```suggestion
        return [extracted] if not isinstance(extracted, list) else extracted
```

File: airbyte-integrations/connectors/source-monday/source_monday/extractor.py [hunk 14]
```diff
@@ -0,0 +1,88 @@
...
+    def try_extract_records(self, response: requests.Response, field_path: List[Union[InterpolatedString, str]]) -> List[Record]:
+        response_body = self.decoder.decode(response)
+        if len(field_path) == 0:
+            extracted = response_body
+        else:
+            path = [path.eval(self.config) for path in field_path]
+            if "*" in path:
+                extracted = dpath.util.values(response_body, path)
+            else:
+                extracted = dpath.util.get(response_body, path, default=[])
```
Comment (lazebnyi) [inline airbyte-integrations/connectors/source-monday/source_monday/extractor.py:76]:
```suggestion
        extracted = response_body  if len(field_path) == 0 else extracted = dpath.util.get(response_body, path, default=[])

        path = [path.eval(self.config) for path in field_path]
        if "*" in path:
            extracted = dpath.util.values(response_body, path)
```

**Review by lazebnyi — CHANGES_REQUESTED** [review]

**Comment by octavia-squidington-iii** [comment]
##  source-monday test report (commit [`1746c57a70`](https://github.com/airbytehq/airbyte/pull/27944/commits/1746c57a70605baad6a2e33a89627648045f4588)) - ❌

⏲️  Total pipeline duration: 57.28s 

| Step                                                                 | Result   |
|:---------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-monday/metadata.yaml | ✅        |
| Connector version semver check                                       | ✅        |
| QA checks                                                            | ✅        |
| Code format checks                                                   | ✅        |
| Connector package install                                            | ✅        |
| Build source-monday docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                           | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/tolik0_source-monday_20950-source-mondaycom-incremental-sync/1689096435/1746c57a70605baad6a2e33a89627648045f4588/source-monday/1.1.0/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-monday test
```

**Comment by octavia-squidington-iii** [comment]
##  source-monday test report (commit [`df87dd0970`](https://github.com/airbytehq/airbyte/pull/27944/commits/df87dd09704284f36ade9f5f95658058558c8c4e)) - ❌

⏲️  Total pipeline duration: 36.09s 

| Step                                                                 | Result   |
|:---------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-monday/metadata.yaml | ✅        |
| Connector version semver check                                       | ✅        |
| QA checks                                                            | ✅        |
| Code format checks                                                   | ✅        |
| Connector package install                                            | ✅        |
| Build source-monday docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                           | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/tolik0_source-monday_20950-source-mondaycom-incremental-sync/1689097147/df87dd09704284f36ade9f5f95658058558c8c4e/source-monday/1.1.0/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-monday test
```

**Comment by octavia-squidington-iii** [comment]
##  source-monday test report (commit [`19cbfa94ca`](https://github.com/airbytehq/airbyte/pull/27944/commits/19cbfa94cad88980cc926eea9e354ea82b8d112d)) - ✅

⏲️  Total pipeline duration: 04mn08s 

| Step                                                                 | Result   |
|:---------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-monday/metadata.yaml | ✅        |
| Connector version semver check                                       | ✅        |
| QA checks                                                            | ✅        |
| Code format checks                                                   | ✅        |
| Connector package install                                            | ✅        |
| Build source-monday docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                           | ✅        |
| Acceptance tests                                                     | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/tolik0_source-monday_20950-source-mondaycom-incremental-sync/1689097883/19cbfa94cad88980cc926eea9e354ea82b8d112d/source-monday/1.1.0/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-monday test
```

File: airbyte-integrations/connectors/source-monday/source_monday/extractor.py (hunk 13 above)
Comment (tolik0) [reply airbyte-integrations/connectors/source-monday/source_monday/extractor.py:90]: Suggested changes will not return the same value

File: airbyte-integrations/connectors/source-monday/source_monday/components.py (hunk 12 above)
Comment (tolik0) [reply airbyte-integrations/connectors/source-monday/source_monday/components.py]: Done

File: airbyte-integrations/connectors/source-monday/source_monday/extractor.py (hunk 14 above)
Comment (tolik0) [reply airbyte-integrations/connectors/source-monday/source_monday/extractor.py:84]: Suggested changes will not return the same value

File: airbyte-integrations/connectors/source-monday/acceptance-test-config.yml [hunk 15]
```diff
@@ -19,16 +19,18 @@ acceptance_tests:
     tests:
       - config_path: "secrets/config.json"
         # `boards`, `items`, `updates` streams schemas were modified
-        # when `type: array` is used, the `items` should be set, instead of `properties`.
-        # this change applies for all configs bellow.
+        #  changed type of id for assets in items and updates
+        #  change deprecated owner to owners in boards
+        # All changes described in corresponding pr
```
[the remark this answers is not in the document]
Comment (tolik0) [reply airbyte-integrations/connectors/source-monday/acceptance-test-config.yml]: Done

File: airbyte-integrations/connectors/source-monday/source_monday/extractor.py (hunk 14 above)
Comment (lazebnyi) [reply airbyte-integrations/connectors/source-monday/source_monday/extractor.py:84]: Please use suggestions as pseudo-code as we discussed. And decrease `if...elif..else` statements.

File: airbyte-integrations/connectors/source-monday/source_monday/extractor.py (hunk 13 above)
Comment (lazebnyi) [reply airbyte-integrations/connectors/source-monday/source_monday/extractor.py:90]: The same as above

**Review by lazebnyi — CHANGES_REQUESTED** [review]

File: airbyte-integrations/connectors/source-monday/acceptance-test-config.yml [hunk 16]
```diff
@@ -18,10 +19,7 @@ acceptance_tests:
   discovery:
     tests:
       - config_path: "secrets/config.json"
-        # `boards`, `items`, `updates` streams schemas were modified
-        #  changed type of id for assets in items and updates
-        #  change deprecated owner to owners in boards
-        # All changes described in corresponding pr
+        # `boards`, `items`, `updates` streams schemas were modified. Pr 27410.
```
Comment (lazebnyi) [inline airbyte-integrations/connectors/source-monday/acceptance-test-config.yml:22]: please add link to PR instead number

File: airbyte-integrations/connectors/source-monday/acceptance-test-config.yml (hunk 15 above)
[the remark this answers is not in the document]
Comment (lazebnyi) [reply airbyte-integrations/connectors/source-monday/acceptance-test-config.yml]: please add link to PR instead number

File: airbyte-integrations/connectors/source-monday/acceptance-test-config.yml (hunk 16 above)
Comment (tolik0) [reply airbyte-integrations/connectors/source-monday/acceptance-test-config.yml]: Done

File: airbyte-integrations/connectors/source-monday/source_monday/extractor.py (hunk 14 above)
Comment (tolik0) [reply airbyte-integrations/connectors/source-monday/source_monday/extractor.py]: Done

File: airbyte-integrations/connectors/source-monday/source_monday/extractor.py (hunk 13 above)
Comment (tolik0) [reply airbyte-integrations/connectors/source-monday/source_monday/extractor.py]: Done

**Comment by octavia-squidington-iii** [comment]
##  source-monday test report (commit [`4c31c40767`](https://github.com/airbytehq/airbyte/pull/27944/commits/4c31c40767e773a215bc21a4305e78e6f4972cd5)) - ✅

⏲️  Total pipeline duration: 04mn14s 

| Step                                                                 | Result   |
|:---------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-monday/metadata.yaml | ✅        |
| Connector version semver check                                       | ✅        |
| QA checks                                                            | ✅        |
| Code format checks                                                   | ✅        |
| Connector package install                                            | ✅        |
| Build source-monday docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                           | ✅        |
| Acceptance tests                                                     | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/tolik0_source-monday_20950-source-mondaycom-incremental-sync/1689122620/4c31c40767e773a215bc21a4305e78e6f4972cd5/source-monday/1.1.0/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-monday test
```

File: airbyte-integrations/connectors/source-monday/acceptance-test-config.yml [hunk 17]
```diff
@@ -18,10 +19,7 @@ acceptance_tests:
   discovery:
     tests:
       - config_path: "secrets/config.json"
-        # `boards`, `items`, `updates` streams schemas were modified
-        #  changed type of id for assets in items and updates
-        #  change deprecated owner to owners in boards
-        # All changes described in corresponding pr
+        # `boards`, `items`, `updates` streams schemas were modified. Pr: https://github.com/airbytehq/airbyte/pull/27410
```
Comment (lazebnyi) [inline airbyte-integrations/connectors/source-monday/acceptance-test-config.yml:22]: PR better write in upper case as acronym.

**Review by lazebnyi — APPROVED** [review]
LGTM! Minor change required.

File: airbyte-integrations/connectors/source-monday/acceptance-test-config.yml (hunk 17 above)
Comment (tolik0) [reply airbyte-integrations/connectors/source-monday/acceptance-test-config.yml]: Done

**Comment by octavia-squidington-iii** [comment]
##  source-monday test report (commit [`334138abe9`](https://github.com/airbytehq/airbyte/pull/27944/commits/334138abe946e87e57996504636f9a44f7a1c4a6)) - ✅

⏲️  Total pipeline duration: 04mn11s 

| Step                                                                 | Result   |
|:---------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-monday/metadata.yaml | ✅        |
| Connector version semver check                                       | ✅        |
| QA checks                                                            | ✅        |
| Code format checks                                                   | ✅        |
| Connector package install                                            | ✅        |
| Build source-monday docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                           | ✅        |
| Acceptance tests                                                     | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/tolik0_source-monday_20950-source-mondaycom-incremental-sync/1689172973/334138abe946e87e57996504636f9a44f7a1c4a6/source-monday/1.1.0/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-monday test
```

**Pull request merged**